### ISSUE TITLE
Pull in Orion-BMS-Serial to Core

### ISSUE THREAD

**Issue by EdHurtig** [openloopalliance/hyperloop-core#40]
@matkam11 has done the legwork getting the OrionBMS userspace driver written.  Now it is time to bring that into the main hyperloop repo under libhw

**Comment by EdHurtig** [comment]
@Dan-Hartman let's assume that the source code in the Orion BMS repo is good and port it into libhw anyways. If there are packet formatting issues we can resolve that at a later time.

**Comment by QuantumAlmonds** [comment]
Matan said that the code does not fully work.

**Comment by QuantumAlmonds** [comment]
Just port and fix later?

**Comment by EdHurtig** [comment]
Yeah, we can fix some of the protocol level stuff later, right now having plumbing for `get_current` `get_voltage` etc to pipe into the `batttery_pack_t` structs already in core would be beneficial

**Comment by EdHurtig** [comment]
For reference: OBD2 is implemented on the serial interface https://www.orionbms.com/general/retrieving-data-obd2-canbus/

**Comment by EdHurtig** [comment]
https://learn.sparkfun.com/tutorials/getting-started-with-obd-ii

**Comment by QuantumAlmonds** [comment]
👍 on it!
